cosmo_tester/framework/cluster.py

```diff
@@ -638,17 +641,17 @@ def _bootstrap_managers(self):
         super(BootstrapBasedCloudifyCluster, self)._bootstrap_managers()
 
         self._clone_manager_blueprints()
-        self._create_inputs_file()
-        self._bootstrap_manager()
+        for manager in self.managers:
+            self._create_inputs_file(manager)
+            self._bootstrap_manager()
```

Comment:
this is very ugly because this depends on `_create_inputs_file` setting an instance attribute (inputs), and `_bootstrap_manager` reading that.
Perhaps make the inputs file a return from the first function, and an argument to the latter. Also of course would need to make the filename non-constant.
This would be a first step towards having parallel (threaded) bootstraps. (the next step would be having separate CFY_WORKDIRs)